### PR TITLE
[Julia] Add concept exercise: leap

### PR DESCRIPTION
**Pull request by SaschaMann**

### CODE REVIEW

File: languages/julia/exercises/concept/leap/.docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Given a year, report if it is a leap year in the Gregorian calendar.
+
+## Arithmetic Operators: Remainder & Equality
+
+Julia provides a number of [arithmetic operators](https://en.wikipedia.org/wiki/Arithmetic#Arithmetic_operations) for primitive numeric types[^1].
```
Comment (SaschaMann) [inline languages/julia/exercises/concept/leap/.docs/introduction.md:5]:
```suggestion
Julia provides a number of [arithmetic operators](https://en.wikipedia.org/wiki/Arithmetic#Arithmetic_operations) for numeric types[^1].
```

File: languages/julia/exercises/concept/leap/.docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+Given a year, report if it is a leap year in the Gregorian calendar.
+
+## Arithmetic Operators: Remainder & Equality
+
+Julia provides a number of [arithmetic operators](https://en.wikipedia.org/wiki/Arithmetic#Arithmetic_operations) for numeric types[^1].
+
+In this exercise you will only need the `%` operator that calculates the [remainder](https://en.wikipedia.org/wiki/Remainder)
```
Comment (SaschaMann) [inline languages/julia/exercises/concept/leap/.docs/introduction.md:7]:
```suggestion
In this exercise you will only need the `%` operator that calculates the [remainder](https://en.wikipedia.org/wiki/Remainder) of the division of two [natural numbers](https://en.wikipedia.org/wiki/Natural_numbers)
```

File: languages/julia/exercises/concept/leap/.docs/introduction.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ Given a year, report if it is a leap year in the Gregorian calendar.
 
 Julia provides a number of [arithmetic operators](https://en.wikipedia.org/wiki/Arithmetic#Arithmetic_operations) for numeric types[^1].
 
-In this exercise you will only need the `%` operator that calculates the [remainder](https://en.wikipedia.org/wiki/Remainder)
+In this exercise you will only need the `%` operator that calculates the [remainder](https://en.wikipedia.org/wiki/Remainder) of the division of two [natural numbers](https://en.wikipedia.org/wiki/Natural_numbers)
```
Comment (cmcaine) [inline languages/julia/exercises/concept/leap/.docs/introduction.md:7]: % works with floats (badly), negative numbers, etc.

Comment (SaschaMann) [reply languages/julia/exercises/concept/leap/.docs/introduction.md:7]: I know but it's not relevant to this exercise.

Would you prefer this?

```suggestion
In this exercise you will only need the `%` operator that calculates the [remainder](https://en.wikipedia.org/wiki/Remainder) of the division of two numbers
```

Comment (cmcaine) [reply languages/julia/exercises/concept/leap/.docs/introduction.md:7]: I think so? I just worry we'll confuse people if we say it's for natural numbers.

**Comment by cmcaine** [comment]
Predicate is maybe jargon?

**Comment by SaschaMann** [comment]
Can you think of a synonym that is better? The ones that I found are also potentially jargon.

**Comment by cmcaine** [comment]
I think we should say something like:

> predicates are statements or functions that are either true or false,
depending on their inputs. `isodd(x)` is an example of a predicate function
is Julia

**Comment by SaschaMann** [comment]
I like that, added.

**Review by iHiD — APPROVED** [review]

**Pull request merged**